react/features/base/session/actions.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+
+import { conferenceLeft, conferenceWillLeave } from '../conference';
+
+import { SESSION_CREATED, SESSION_FAILED, SESSION_STARTED, SESSION_TERMINATED } from './actionTypes';
+import { Session } from './Session';
+
+import type { Dispatch } from 'redux';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * FIXME.
+ *
+ * @returns {Function}
+ */
+export function endAllSessions() {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const sessions = getState()['features/base/session'];
+
+        for (const session of sessions.values()) {
+            dispatch(endSession(session));
+        }
+    };
+}
+
+/**
+ * FIXME.
+ *
+ * @param {Session} session - FIXME.
+ * @returns {function(*): *}
+ */
+export function endSession(session: Session) {
+    return (dispatch: Dispatch<any>) => {
+        // The conference we have already joined or are joining.
+        const conference_ = session.conference;
+
+        // Promise which completes when the conference has been left and the connection has been disconnected.
+        let promise;
+
+        // Leave the conference.
+        if (conference_) {
+            // In a fashion similar to JitsiConference's CONFERENCE_LEFT event (and the respective Redux action) which
+            // is fired after the conference has been left, notify the application about the intention to leave
+            // the conference.
+            dispatch(conferenceWillLeave(conference_));
+
+            promise
+                = conference_.leave()
+                    .catch(error => {
+                        logger.warn('JitsiConference.leave() rejected with:', error);
+
+                        // The library lib-jitsi-meet failed to make the JitsiConference leave. Which may be because
+                        // JitsiConference thinks it has already left. Regardless of the failure reason, continue in
+                        // jitsi-meet as if the leave has succeeded.
+                        dispatch(conferenceLeft(conference_));
+                    });
+        } else {
+            promise = Promise.resolve();
+        }
+
+        const connection_ = session.connection;
+
+        if (connection_) {
```

Comment:
No, in that case things will end up on the "base/config" level with `LOAD_CONFIG_FAILED` dispatched and this will clear the session. It will be dispatched, because we will navigate to new location and the LOAD_CONFIG request will become outdated and discarded at `appNavigate` around line 106.

Reply:
So actually not that, but there's this event `endAllSessions` dispatched at the beginning of `appNavigate`. But the config thing could also work.

Reply:
I'd like us to got for explicit behavior, not implicit. Chasing how a connection is closed by following the appNavigate route is kinda backwards IMHO.